mail-connector/src/test/java/eu/tsystems/mms/tic/testframework/mailconnector/test/MailConnectorTest.java

```diff
@@ -703,7 +718,14 @@ private Email waitForMessage(final SearchTerm searchTerm, final AbstractInboxCon
 
         // TEST - Fail, if no message was received.
         try {
-            receivedMsg = abstractInboxConnector.waitForMails(searchTerm, folderName).get(0);
+            final EmailQuery query = new EmailQuery()
+                    .setSearchTerm(searchTerm)
+                    .setFolderName(folderName);
+            final List<Email> emailList = abstractInboxConnector.query(query).collect(Collectors.toList());
+            if (emailList.isEmpty()) {
```

Comment:
Please use an `Assert` to check for `emailList.isEmpty`. I think the `try...catch` is not needed here.